### ISSUE TITLE
Codership's galera .DEB package can not be used with XtraDB Cluster without foricing

### ISSUE THREAD

**Issue by philip-galera** [codership/galera#231]
When installing the XtraDB Cluster, it is not possible to avoid installing Percona's own galera library as well (percona-xtradb-cluster-galera-2.x). It seems to be downloaded automatically from the Ubuntu repository.

Then, it is not possible to replace percona-xtradb-cluster-galera-2.x with Codership's galera package:

```
galera@ubuntu14:~$ sudo dpkg -i galera-3_25.3.9.20150219.fa8d959-1trusty_amd64.deb
Selecting previously unselected package galera-3.
dpkg: regarding galera-3_25.3.9.20150219.fa8d959-1trusty_amd64.deb containing galera-3:
 galera-3 conflicts with percona-xtradb-cluster-galera-2.x
  percona-xtradb-cluster-galera-2.x (version 165-0ubuntu1) is present and installed.

dpkg: error processing archive galera-3_25.3.9.20150219.fa8d959-1trusty_amd64.deb (--install):
 conflicting packages - not installing galera-3
Errors were encountered while processing:
 galera-3_25.3.9.20150219.fa8d959-1trusty_amd64.deb
```

So galera-3 is not seen as an upgrade to percona-xtradb-cluster-galera-2.x . It may be because of version numbers:

```
galera@ubuntu14:~$ dpkg -l percona-xtradb-cluster-galera-2.x
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                        Version                    Architecture               Description
+++-===========================================-==========================-==========================-============================================================================================
ii  percona-xtradb-cluster-galera-2.x           165-0ubuntu1               amd64                      Synchronous multi-master replication plugin for transactional applications

vs.

galera@ubuntu14:~$ dpkg -l galera-3
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                        Version                    Architecture               Description
+++-===========================================-==========================-==========================-============================================================================================
ii  galera-3                                    25.3.9.20150219.fa8d959-1t amd64                      Replication framework for transactional applications
```

**Comment by philip-galera** [comment]
Otto, is this something that we can fix on our end, or it requires changes in Percona XtraDB cluster's package definitions?

**Comment by ottok** [comment]
Confirmation:

On a system with 'percona-xtradb-cluster-galera-3' and ''percona-xtradb-cluster-galera-3.x'  installed from...
```
deb http://repo.percona.com/apt trusty main
```
..I added the repository
```
deb http://labs.seravo.fi/~otto/mariadb-repo/ galera-trusty-amd64/
```
...and galera-3 installed nicely:
```
$ apt-get install galera-3
The following packages will be REMOVED:
  percona-xtradb-cluster-galera-3 percona-xtradb-cluster-galera-3.x
The following NEW packages will be installed:
  galera-3
0 upgraded, 1 newly installed, 2 to remove and 388 not upgraded.
Need to get 0 B/784 kB of archives.
After this operation, 65.5 kB disk space will be freed.
Do you want to continue? [Y/n] 
WARNING: The following packages cannot be authenticated!
  galera-3
Install these packages without verification? [y/N] y
Removing percona-xtradb-cluster-galera-3 (3.8.3390.trusty) ...
Removing percona-xtradb-cluster-galera-3.x (3.8.3390.trusty) ...
Selecting previously unselected package galera-3.
(Reading database ... 200746 files and directories currently installed.)
Preparing to unpack .../galera-3_25.3.6-1_amd64.deb ...
Unpacking galera-3 (25.3.6-1) ...
Setting up galera-3 (25.3.6-1) ...
$
```

See comments in #232 - this is the backwards compatibility and upgrade path we want to provide. I would be dangerous to try to replace any current Percona Galera version with Codership Galera version, as they are not the same. It is up to the Percona guys to start depending on the galera-3 package in future versions of their server if they want.

**Comment by ottok** [comment]
I wasn't able to test Percona Cluster 5.5. right now to see how well it upgrades to galera-3 package from their own -2x package, but I am fairly confident I fixed this long time ago and latest by PR#381 this should be fixed now and can be closed.
